util/perl/OpenSSL/config.pm

```diff
@@ -0,0 +1,801 @@
+                # __LP64__ has slipped through, it therefore is not defined
+            } else {
+                $GCC_BITS = '64';
+            }
+        }
+        return;
+    }
+
+    if ( ${SYSTEM} eq 'AIX' ) {
+        # favor vendor cc over gcc
+        if ( okrun('(cc) 2>&1',
+                'grep -iv "not found" >/dev/null') ) {
+            $CC = 'cc';
+        }
+        return;
+    }
+
+    if ( $SYSTEM eq "SunOS" ) {
+        if ( $GCCVER >= 30000 ) {
+            # 64-bit ABI isn't officially supported in gcc 3.0, but seems
+            # to be working; at the very least 'make test' passes.
+            if ( okrun('gcc -v -E -x c /dev/null 2>&1',
+                    'grep __arch64__ >/dev/null') ) {
+                $GCC_ARCH = "-m64"
+            } else {
+                $GCC_ARCH = "-m32"
+            }
+        }
+        # check for WorkShop C, expected output is "cc: blah-blah C x.x"
+        my $v = `(cc -V 2>&1) 2>/dev/null | egrep -e '^cc: .* C [0-9]\.[0-9]'`;
+        chomp $v;
+        $v =~ s/.* C \([0-9]\)\.\([0-9]\).*/$1.$2/;
+        my @numbers = split /\./, $v;
+        my @factors = (10000, 100, 1);
+        while (@numbers && @factors) {
+            $CCVER += pop(@numbers) * pop(@factors)
+        }
+        if ( $MACHINE ne 'i86pc' && $CCVER > 40 ) {
+            # overrides gcc!!!
+            $CC = 'cc';
+            if ( $CCVER == 50000 ) {
+                print <<'EOF';
+WARNING! Found WorkShop C 5.0.
+         Make sure you have patch #107357-01 or later applied.
+EOF
+                maybe_abort();
+            }
+        }
+    }
+}
+
+my $map_patterns =
+    [ [ 'uClinux.*64.*',          { target => 'uClinux-dist64' } ],
+      [ 'uClinux.*',              { target => 'uClinux-dist' } ],
+      [ 'mips3-sgi-irix',         { target => 'irix-mips3' } ],
+      [ 'mips4-sgi-irix64',
+        sub {
+            print <<EOF;
+WARNING! To build 64-bit package, do this:
+         $WHERE/Configure irix64-mips4-$CC
+EOF
+            maybe_abort();
+            return { target => "irix-mips3" };
+        } ],
+      [ 'ppc-apple-rhapsody',     { target => "rhapsody-ppc" } ],
+      [ 'ppc-apple-darwin.*',
+        sub {
+            my $ISA64 = `sysctl -n hw.optional.64bitops 2>/dev/null`;
+            if ( $ISA64 == 1 && $KERNEL_BITS eq '' ) {
+                print <<EOF;
+WARNING! To build 64-bit package, do this:
+         $WHERE/Configure darwin64-ppc-cc
+EOF
+                maybe_abort();
+            }
+            return { target => "darwin64-ppc" }
+                if $ISA64 == 1 && $KERNEL_BITS eq '64';
+            return { target => "darwin-ppc" };
+        } ],
+      [ 'i.86-apple-darwin.*',
+        sub {
+            my $ISA64 = `sysctl -n hw.optional.x86_64 2>/dev/null`;
+            if ( $ISA64 == 1 && $KERNEL_BITS eq '' ) {
+                print <<EOF;
+WARNING! To build 64-bit package, do this:
+         KERNEL_BITS=64 $WHERE/config $options
```

Comment:
shouldn't all of the WARNING! texts here include the $options ?

Reply:
Ah, thanks...  not about `$options`, but `/config` should be changed to `/Configure`

That being said, I'm not sure we need to mention all the options.  There are number of caveats there, among others displaying options that have sh expandable characters...  the temptation to cut-n-paste makes that dicy, to say the least

Reply:
That makes sense.  Maybe put "[options...]" at the end rather than explicitly showing $options?  But at least be consistent. :)

Reply:
I did exactly that